p2p/router.go

```diff
@@ -480,16 +479,37 @@ func (r *Router) sendPeer(peerID NodeID, conn Connection, queue queue) error {
 	}
 }
 
-// SubscribePeerUpdates creates a new peer updates subscription. The caller must
-// consume the peer updates in a timely fashion and close the subscription when
-// done, since delivery is guaranteed and will block peer
-// connection/disconnection otherwise.
-//
-// FIXME: Consider having callers just use peerManager.Subscribe() directly, if
-// we export peerManager and make it an injected dependency (which we probably
-// should).
-func (r *Router) SubscribePeerUpdates() *PeerUpdatesCh {
-	return r.peerManager.Subscribe()
+// evictPeers evicts connected peers as requested by the peer manager.
+func (r *Router) evictPeers() {
+	for {
+		select {
+		case <-r.stopCh:
+			return
+		default:
+		}
+
+		peerID, err := r.peerManager.EvictNext()
+		if err != nil {
+			r.logger.Error("failed to find next peer to evict", "err", err)
+			return
```

Comment:
I'm a bit confused as to why this function returns here. If we get an `error` it's because `peerStore#List` returned an error, but do we really want return here instead of just `continue` with a sleep maybe?

Reply:
Yeah, that's sort of a general problem -- if we encounter an unexpected error, should we abort or try to recover? For safety, I think it's better to abort.

Reply:
Yeah because if that errors and we return, we'll never evict again. But I guess since it error'd, future invocations also won't work?

Reply:
Yup. I think it's fair to say that an unexpected error here typically implies a disk IO error, and there's no point in trying to continue, but we should probably propagate this upwards. However, I don't think there's currently any way for `Service` implementations to signal error to the main process.